container/build.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import argparse
+import os
+import subprocess
+
+# Map to store all supported container type and the package of target to build it.
+TYPE_PACKAGE_MAP = {
+    "rbe-debian8": "container/debian8/builders/rbe-debian8",
+    "rbe-debian9": "container/experimental/rbe-debian9",
+    "rbe-ubuntu16_04": "container/ubuntu16_04/builders/rbe-ubuntu16_04",
+    "ubuntu16_04-bazel": "container/ubuntu16_04/builders/bazel",
+    "ubuntu16_04-bazel-docker": "container/ubuntu16_04/builders/bazel",
+}
+
+# Map to store all supported container type and the name of target to build it.
+TYPE_TARGET_MAP = {
+    "rbe-debian8": "toolchain",
+    "rbe-debian9": "toolchain",
+    "rbe-ubuntu16_04": "toolchain",
+    "ubuntu16_04-bazel": "bazel",
+    "ubuntu16_04-bazel-docker": "bazel_docker",
+}
+
+# Map to store all supported container type and the name of target to build it.
+TYPE_TARBALL_MAP = {
+    "rbe-debian8": "toolchain-packages.tar",
+    "rbe-debian9": "toolchain-packages.tar",
+    "rbe-ubuntu16_04": "toolchain-packages.tar",
+    "ubuntu16_04-bazel": "bazel-packages.tar",
+    "ubuntu16_04-bazel-docker": "bazel_docker-packages.tar",
+}
+
+
+def main(type_, project, container, tag, async_, bucket, local):
```

Comment:
It looks like the `main` function is quite long - we bundled all the functionalities into it. Could we separate it into several smaller functions and `main` is only to coordinate them?

Reply:
Ok, split it into  cloud_build and local_build  functions